src/routes/safe/components/Apps/components/AppFrame.tsx

```diff
@@ -78,9 +80,15 @@ const INITIAL_CONFIRM_TX_MODAL_STATE: ConfirmTransactionModalState = {
   params: undefined,
 }
 
-const safeAppWeb3Provider = new Web3.providers.HttpProvider(getSafeAppsRpcServiceUrl(), {
-  timeout: 10_000,
-})
+const safeAppWeb3Providers: Record<ChainId, HttpProvider> = {}
```

Comment:
Why do we want to keep a reference to previously initialized providers?

Reply:
initializing a provider is not an expensive operation, so I think we shouldn't keep a reference to previously created ones in memory